### PR TITLE
Add Replicated database names to ZooKeeper for introspection

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Databases/DatabaseReplicated.cpp [hunk 1]
```diff
@@ -465,6 +465,13 @@ void DatabaseReplicated::tryConnectToZooKeeperAndInitDatabase(LoadingStrictnessL
             return;
         }
 
+        /// If not exist, create a node with the database name for introspection.
+        /// Technically, the database may have different names on different replicas, but this is not a usual case and we only save the first one
+        auto db_name_path = fs::path(zookeeper_path) / "first_replica_database_name";
```
Comment (SmitaRKulkarni) [inline src/Databases/DatabaseReplicated.cpp:470]: Maybe better to define this in a const "first_replica_database_name"

Comment (tavplubix) [reply src/Databases/DatabaseReplicated.cpp:470]: maybe

**Review by SmitaRKulkarni — APPROVED** [review]
LGTM

**Comment by tavplubix** [comment]
Some fixes for CI failures: https://github.com/ClickHouse/ClickHouse/pull/65720, https://github.com/ClickHouse/ClickHouse/pull/65746

**Pull request merged**